### PR TITLE
RoleChanged to transpilers

### PR DESCRIPTION
**Pull request by Ficus-x**
xD

### CODE REVIEW

**Comment by joker-119** [comment]
Ahh yes, adding a ret instruction to the end of the method is totally the same thing.

Learn how transpilers work.

**Comment by IRacle1** [comment]
🙃

File: Exiled.Events/Patches/Fixes/RoleChangedPatch.cs [hunk 1]
```diff
@@ -7,16 +7,30 @@
...
+    using NorthwoodLib.Pools;
 
     /// <summary>
     /// Patches <see cref="InventoryItemProvider.RoleChanged"/> to help override in <see cref="EventArgs.ChangingRoleEventArgs.Items"/> and <see cref="EventArgs.ChangingRoleEventArgs.Ammo"/>.
     /// </summary>
     [HarmonyPatch(typeof(InventoryItemProvider), nameof(InventoryItemProvider.RoleChanged))]
     internal static class RoleChangedPatch
     {
-        private static bool Prefix() => false;
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Fixes/RoleChangedPatch.cs:23]: `private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions) => yield return OpCodes.Ret;` or something

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Fixes/RoleChangedPatch.cs (hunk 1 above)
Comment (Ficus-x) [reply Exiled.Events/Patches/Fixes/RoleChangedPatch.cs:23]: Solved

**Review by Michal78900 — APPROVED** [review]

**Pull request merged**